File: csharp/src/Ice/SSL/Instance.cs
// Copyright (c) ZeroC, Inc.

using System.Net.Security;
using System.Security.Cryptography.X509Certificates;

namespace Ice.SSL;

internal class Instance : Ice.Internal.ProtocolInstance
{
    internal Instance(SSLEngine engine, short type, string protocol)
        : base(engine.communicator(), type, protocol, true) => _engine = engine;

    internal SSLEngine engine() => _engine;

    internal int securityTraceLevel() => _engine.securityTraceLevel();

    internal string securityTraceCategory() => _engine.securityTraceCategory();

    internal X509Certificate2Collection certs() => _engine.certs();

    internal void traceStream(SslStream stream, string connInfo) => _engine.traceStream(stream, connInfo);

    internal void verifyPeer(ConnectionInfo info, string description) =>
        _engine.verifyPeer(info, description);

    internal Ice.InitializationData initializationData() =>
        _engine.communicator().instance.initializationData();

    private readonly SSLEngine _engine;
}

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/SSL/Instance.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice.SSL;

import com.zeroc.Ice.InitializationData;
import com.zeroc.Ice.ProtocolInstance;

/**
 * @hidden Public because it's used by Ice.Instance
 */
public class Instance extends ProtocolInstance {
    public Instance(SSLEngine engine, short type, String protocol) {
        super(engine.communicator(), type, protocol, true);
        _engine = engine;
    }

    SSLEngine engine() {
        return _engine;
    }

    int securityTraceLevel() {
        return _engine.securityTraceLevel();
    }

    void traceConnection(String desc, javax.net.ssl.SSLEngine engine, boolean incoming) {
        _engine.traceConnection(desc, engine, incoming);
    }

    void verifyPeer(String address, ConnectionInfo info, String desc) {
        _engine.verifyPeer(address, info, desc);
    }

    InitializationData initializationData() {
        return _engine.communicator().getInstance().initializationData();
    }

    private final SSLEngine _engine;
}
